### DISCUSSION TITLE
centos7无法运行bpftrace和bpftrace tools

### DISCUSSION THREAD

**Discussion by Scroog** [bpftrace/bpftrace#3085] [General]
是的，我的系统是centos7，但是无法运行bpftrace,相关告警如下：
![image](https://github.com/bpftrace/bpftrace/assets/91512674/cf2b17b5-e98a-41b5-9079-66cbaa2852c3)
![屏幕截图 2024-03-22 184234](https://github.com/bpftrace/bpftrace/assets/91512674/a6af3174-72ef-4d9a-9354-1aeed74e265a)
[root@localhost bpf]# bpftrace -v -e 'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'
INFO: node count: 7
Attaching 1 probe...

Error log:

ERROR: Error loading program: tracepoint:raw_syscalls:sys_enter
目前我已经使用了root权限运行。

**Comment by viktormalik** [comment]
What bpftrace version is this?

**Reply by Scroog** [reply]
最新的版本，v0.20.2

**Reply by Scroog** [reply]
ccentos7的内核版本为3.10.0

**Reply by Scroog** [reply]
bpftrace不能在这个版本的centos7上运行？

**Reply by viktormalik** [reply]
bpftrace 0.20.2 requires a kernel with BTF support which CentOS 7 does not have. The only option is to use an older bpftrace, e.g. 0.19.

**Reply by Scroog** [reply]
好的，我去试试  非常感谢

**Reply by Scroog** [reply]
[root@localhost bpf]# ./bpftrace
./bpftrace: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by ./bpftrace)
./bpftrace: /lib64/libc.so.6: version `GLIBC_2.27' not found (required by ./bpftrace)
./bpftrace: /lib64/libc.so.6: version `GLIBC_2.22' not found (required by ./bpftrace)
  现在出现了这个错误，使用的版本是0.19.0

**Reply by viktormalik** [reply]
It's quite tough to navigate in the Chinese text but it seems that there's a conflict between the installed glibc version and the one bpftrace has been compiled against. I'd recommend compiling bpftrace locally.

**Reply by Scroog** [reply]
这个我是直接下载的可执行程序，那么在本地编译也是编译0.19版本的吗
